File: scala/myfunctions.py
# Databricks notebook source
# MAGIC %scala
# MAGIC import org.apache.spark.sql.DataFrame
# MAGIC import org.apache.spark.sql.functions.col
# MAGIC
# MAGIC // Does the specified table exist in the specified database?
# MAGIC def tableExists(tableName: String, dbName: String) : Boolean = {
# MAGIC   return spark.catalog.tableExists(dbName + "." + tableName)
# MAGIC }
# MAGIC
# MAGIC // Does the specified column exist in the given DataFrame?
# MAGIC def columnExists(dataFrame: DataFrame, columnName: String) : Boolean = {
# MAGIC   val nameOfColumn = null
# MAGIC
# MAGIC   for(nameOfColumn <- dataFrame.columns) {
# MAGIC     if (nameOfColumn == columnName) {
# MAGIC       return true
# MAGIC     }
# MAGIC   }
# MAGIC
# MAGIC   return false
# MAGIC }
# MAGIC
# MAGIC // How many rows are there for the specified value in the specified column
# MAGIC // in the given DataFrame?
# MAGIC def numRowsInColumnForValue(dataFrame: DataFrame, columnName: String, columnValue: String) : Long = {
# MAGIC   val df = dataFrame.filter(col(columnName) === columnValue)
# MAGIC
# MAGIC   return df.count()
# MAGIC }

File: R/myfunctions.r
library(SparkR)

# Does the specified table exist in the specified database?
table_exists <- function(table_name, db_name) {
  tableExists(paste(db_name, ".", table_name, sep = ""))
}

# Does the specified column exist in the given DataFrame?
column_exists <- function(dataframe, column_name) {
  column_name %in% colnames(dataframe)
}

# How many rows are there for the specified value in the specified column
# in the given DataFrame?
num_rows_in_column_for_value <- function(dataframe, column_name, column_value) {
  df = filter(dataframe, dataframe[[column_name]] == column_value)

  count(df)
}